src/Agent.Worker/TaskCommandExtension.cs

```diff
@@ -232,6 +236,33 @@ private void ProcessTaskUploadSummaryCommand(IExecutionContext context, string d
             }
         }
 
+        private void ProcessTaskUploadFileCommand(IExecutionContext context, string data)
+        {
+            if (!String.IsNullOrEmpty(data))
+            {
+                var filePath = Path.GetFullPath(data);
+
+                if (File.Exists(filePath))
+                {
+                    var fileName = Path.GetFileName(filePath);
+                    var properties = new Dictionary<string, string>();
+
+                    properties.Add(TaskAddAttachmentEventProperties.Type, CoreAttachmentType.FileAttachment);
+                    properties.Add(TaskAddAttachmentEventProperties.Name, fileName);
+
+                    ProcessTaskAddAttachmentCommand(context, properties, filePath);
+                }
+                else
+                {
+                    throw new Exception(string.Format(CultureInfo.InvariantCulture, "Unable to find file to upload from: {0}.", filePath));
+                }
+            }
+            else
+            {
+                throw new Exception("Cannot upload file because file location is not specified.");
```

Comment:
These two new strings should be localized. Can you add the string to `/src/Misc/layoutbin/en-US/strings.json` and use `StringUtil.Loc(key[, arg...])`?

It looks like the exception message in the method above isn't localized either (existing bug). Can you please add a TODO comment in that location (or go ahead and move it to strings.json as well if you feel comfortable with that change)?